### DISCUSSION TITLE
Automatically closing stale issues and PRs

### DISCUSSION THREAD

**Discussion by greglandrum** [rdkit/rdkit#6022] [Development]
I'd like to add a Github Action to the RDKit repo to automatically close issues and PRs which have not been updated in a long time. This one: https://github.com/marketplace/actions/close-stale-issues first marks things as "stale" if they haven't been updated in 60 days and then closes them after 7 additional days of inactivity. Any activity will result in a "stale" label being removed and we can always re-open things if they become active again.

Any objections to this or suggestions for better solutions?

@bp-kelley @ricrogz @ptosco , as fellow maintainers, I'd be particularly interested in your thoughts, or just a thumbs up if you agree with the idea.

**Comment by mwojcikowski** [comment]
Hi Greg, I would argue that closing bugs may end up counter-productive and lead to more duplicates. At the end of the day they are still there, and having an open bug sends a clear message - it was not fixed.

My suggestion would be to make an exception for anything that has a bug label on it, which is usually added/removed by the maintainer when doing the initial tirage of issues.

**Reply by greglandrum** [reply]
> My suggestion would be to make an exception for anything that has a bug label on it, which is usually added/removed by the maintainer when doing the initial tirage of issues.

[the first draft of this sounded dismissive, which I didn't intend to be, so I've edited the reply]

That's useful feedback and I think you make good points. Unfortunately the bug labels are a mess since they aren't assigned by a moderator but are added when the issue is created. Still, if we  did exclude bugs (this is possible), we could still at least somewhat reduce the giant backlog.

About the number of duplicates. It's hard to be sure, but given the number of duplicates we get compared to the number of "this happens to me too" type comments on old issues I don't think that a lot of people are spending time searching the issue tracker as it stands.

**Comment by ptosco** [comment]
@greglandrum There is no perfect solution here. Well, there would be one: going through the list of open PRs and bugs, triaging and closing as appropriate, removing duplicates, fixing the tons of conflicts in half-done PRs and bringing them to completion... I'd love to be able to do all of that, and it causes frustration and pain not to have any time to dedicate.
I think your pragmatic solution is a good one to tidy up the backlog: after all, this is hardly going to impact critical issues (otherwise they would have been addressed more timely). And closed issues/PRs can always be searched and reopened (I do remember a few stale PRs of mine I need to complete, and I will look for them when I will, eventually, find the time).
So I would go ahead with your proposed solution,

**Comment by ricrogz** [comment]
In a general sense, I think marking old issues/PRs as "stale" and closing them if they don't see activity is a good idea. Not having a huge backlog should help us focus on the things that are open.

There's a couple of things I'm not quite sure how they will go, like, if people will ping their cases to keep them alive when they get a "stale" stamp, or if we'd need some kind of mark to prevent issues being worked on from being marked stale/being close (although I think we are  usually rather fast to work on PRs, and we don't usually work on stuff long enough as to have it marked "stale"). But I think we can see how things go, and make changes in the future if we need it.
